src/fireedge/src/modules/components/Forms/Vm/CreateSchedActionForm/fields.js

```diff
@@ -0,0 +1,610 @@
+      }
+
+      return dayValues[repeatType] ?? value
+    }),
+}
+
+// --------------------------------------------------------
+// End fields
+// --------------------------------------------------------
+
+/** @type {Field} End type field */
+const END_TYPE_FIELD = {
+  name: 'END_TYPE',
+  label: T.EndType,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  optionsOnly: true,
+  dependOf: PERIODIC_FIELD_NAME,
+  htmlType: (typeAction) =>
+    typeAction !== SCHEDULE_TYPE.PERIODIC && INPUT_TYPES.HIDDEN,
+  values: arrayToOptions(Object.keys(END_TYPE_VALUES), {
+    addEmpty: false,
+    getText: (value) => sentenceCase(value),
+    getValue: (value) => END_TYPE_VALUES[value],
+  }),
+  validation: mixed().notRequired(),
+  grid: { md: 12 },
+}
+
+/** @type {Field} End value field */
+const END_VALUE_FIELD = {
+  name: 'END_VALUE',
+  label: ([typeAction, endType] = []) =>
+    typeAction === SCHEDULE_TYPE.PERIODIC && endType === END_TYPE_VALUES.DATE
+      ? T.WhenDoYouWantThisActionToStop
+      : T.HowManyTimesDoYouWantThisActionToExecute,
+  dependOf: [PERIODIC_FIELD_NAME, END_TYPE_FIELD.name],
+  type: ([typeAction, endType] = []) =>
+    typeAction === SCHEDULE_TYPE.PERIODIC && endType === END_TYPE_VALUES.DATE
+      ? INPUT_TYPES.TIME
+      : INPUT_TYPES.TEXT,
+  htmlType: (depends = []) => {
+    const [PERIODIC, END_TYPE] = depends
+
+    return PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+      END_TYPE !== END_TYPE_VALUES.NEVER
+      ? 'number'
+      : INPUT_TYPES.HIDDEN
+  },
+  validation: mixed().when(
+    [PERIODIC_FIELD_NAME, END_TYPE_FIELD.name],
+    (typeAction, endType) => {
+      if (typeAction === SCHEDULE_TYPE.PERIODIC) {
+        return {
+          [END_TYPE_VALUES.NEVER]: string().strip(),
+          [END_TYPE_VALUES.REPETITION]: number().required().min(1),
+          [END_TYPE_VALUES.DATE]: date()
+            .min(ref(TIME_FIELD.name))
+            .required()
+            .transform(parseDateString)
+            .afterSubmit(dateToMilliseconds),
+        }[endType]
+      } else {
+        return string().trim().notRequired()
+      }
+    }
+  ),
+  grid: { md: 12 },
+  fieldProps: ([_, endType] = []) =>
+    endType === END_TYPE_VALUES.DATE && { defaultValue: getNextWeek() },
+}
+
+// --------------------------------------------------------
+// Relative fields
+// --------------------------------------------------------
+
+/** @type {Field} Relative time field */
+export const RELATIVE_TIME_FIELD = {
+  name: 'RELATIVE_TIME',
+  label: T.TimeAfterTheVmIsInstantiated,
+  type: INPUT_TYPES.TEXT,
+  dependOf: PERIODIC_FIELD_NAME,
+  htmlType: (typeAction) =>
+    typeAction === SCHEDULE_TYPE.RELATIVE ? 'number' : INPUT_TYPES.HIDDEN,
+  validation: number()
+    .positive()
+    .transform((value, originalValue) =>
+      isRelative(originalValue)
+        ? getPeriodicityByTimeInSeconds(originalValue)?.time
+        : value
+    )
+    .when(PERIODIC_FIELD_NAME, (typeAction, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).